TelemetryProcessors/SamplingTelemetryProcessor.ts

```diff
@@ -0,0 +1,45 @@
+import ContractsModule = require("../Library/Contracts");
+import { CorrelationContext } from "../AutoCollection/CorrelationContextManager";
+
+/**
+ *  A telemetry processor that handles sampling.
+ */
+export function samplingTelemetryProcessor(envelope: ContractsModule.Contracts.Envelope, contextObjects: { correlationContext: CorrelationContext }): boolean {
+    var samplingPercentage = envelope.sampleRate; // Set for us in Client.getEnvelope
+    var isSampledIn = false;
+
+    if (samplingPercentage === null || samplingPercentage === undefined || samplingPercentage >= 100) {
+        return true;
+    } else if (contextObjects.correlationContext && contextObjects.correlationContext.operation) {
+        // If we're using dependency correlation, sampling should retain all telemetry from a given request 
+        isSampledIn = getSamplingHashCode(contextObjects.correlationContext.operation.id) < samplingPercentage;
```

Comment:
I don't believe we currently keep user id within the correlation context, so that will work for requests but no dependencies. We could update the correlation logic but that's probably a separate PR